OSQuery failed due to error C2428 under /std:c++latest +MSVC

Hi all,

Issue description:
We tried to build OSQuery using VS2019 on Windows Server 2016. OSQuery failed to build due to error C2428: '++': not allowed on operand of type 'bool'. It can be reproduced on release branch latest commit 1c5ed8a. Could you please help take a look at this? Thanks in advance!

[build.log](https://github.com/osquery/osquery/files/5040530/build.log)

**Repro steps:**
1. git clone --recursive https://github.com/facebook/osquery F:\gitP\facebook\osquery
2. open a VS 2019 x64 command prompt and browse to F:\gitP\facebook\osquery
3. set PATH=C:\Python\Python27\;%PATH%
4. set PATH=C:\Python\Python27\Scripts;%PATH%
5. cmake -G "Visual Studio 16 2019" -A x64 -DCMAKE_SYSTEM_VERSION=10.0.18362.0 ..
6. cmake --build . --target osqueryd --config Release --verbose -- /maxcpucount

**Actual result:**
185>F:\gitP\facebook\osquery\osquery\tables\events\windows\powershell_events.cpp(177,26): error C2428: '++': not allowed on operand of type 'bool' [F:\gitP\facebook\osquery\build\osquery\tables\events\osquery_tables_events_eventstable.vcxproj]
185>F:\gitP\facebook\osquery\osquery\tables\events\windows\powershell_events.cpp(186,26): error C2428: '++': not allowed on operand of type 'bool' [F:\gitP\facebook\osquery\build\osquery\tables\events\osquery_tables_events_eventstable.vcxproj]

--- a/osquery/tables/events/windows/powershell_events.cpp
+++ b/osquery/tables/events/windows/powershell_events.cpp
@@ -166,25 +166,26 @@ Status PowershellEventSubscriber::parseScriptMessageEvent(
     auto field_name = node.get("<xmlattr>.Name", "");
     if (field_name.empty()) {
       malformed_field = true;
-      continue;
+      break;
     }
 
     auto field_string_value = node.data();
-    auto field_integer_value_exp = tryTo<std::size_t>(field_string_value);
 
     if (field_name == "MessageNumber") {
+      auto field_integer_value_exp = tryTo<std::size_t>(field_string_value);
       if (field_integer_value_exp.isError()) {
-        ++malformed_field;
-        continue;
+        malformed_field = true;
+        break;
       }
 
       output.message_number = field_integer_value_exp.take();
       ++field_count;
 
     } else if (field_name == "MessageTotal") {
+      auto field_integer_value_exp = tryTo<std::size_t>(field_string_value);
       if (field_integer_value_exp.isError()) {
-        ++malformed_field;
-        continue;
+        malformed_field = true;
+        break;
       }
 
       output.expected_message_count = field_integer_value_exp.take();

